lang/cs/Org.Apache.REEF.IMRU/OnREEF/ResultHandler/WriteResultHandler.cs

```diff
@@ -84,12 +89,31 @@ public void Dispose()
             if (_fileSystem.Exists(remoteUri))
             {
                 Exceptions.Throw(
-                    new Exception(string.Format("Output Uri: {0} already exists", remoteUri)), Logger);
+                    new Exception(string.Format("Output Uri: {0} already exists", remoteUri)),
+                    Logger);
             }
 
             _fileSystem.CopyFromLocal(_localFilename, remoteUri);
         }
 
+        /// <summary>
+        /// Delete local file
+        /// </summary>
+        public void Dispose()
+        {
+            if (_localFilename != null)
+            {
+                lock (_disposeLock)
```

Comment:
`_resultHandler.Dispose()` is called only in `UpdateTaskHost.Dispose()`, which seems to have lock of its own (`Interlocked.Exchange == 0` is supposed to be entered only once, after that it will be only 1). Do we need this lock here?